### PR TITLE
Escape strings when writing json

### PR DESCRIPTION
**Pull request by lennartbuit**

### CODE REVIEW

File: json_emitter.go [hunk 1]
```diff
@@ -51,7 +53,11 @@ func maybePrefix(prefix, tag, str string) string {
 
 func (j *JsonEmitter) emitString(prefix, tag, str string, asMapKey bool, cache WriteCache) error {
 	outString := cache.CacheWrite(maybePrefix(prefix, tag, str), asMapKey)
-	j.buffer.WriteString(fmt.Sprintf("\"%s\"", outString))
+	value, err := json.Marshal(outString)
+	if err != nil {
```
Comment (jkemperman) [inline json_emitter.go:57]: Very idiomatic, bravo

image

**Review by jkemperman — APPROVED** [review]
Nice!

File: writer_test.go [hunk 2]
```diff
@@ -39,6 +39,11 @@ var _ = Describe("JSON Writer", func() {
 		Expect(result).To(Equal("[\"~#'\",\"a string\"]"))
 	})
 
+	It("marshals strings should be escaped", func() {
```
Comment (lennartbuit) [inline writer_test.go]:
```suggestion
	It("marshals strings which should be escaped", func() {
```

File: writer_test.go [hunk 3]
```diff
@@ -39,6 +39,11 @@ var _ = Describe("JSON Writer", func() {
 		Expect(result).To(Equal("[\"~#'\",\"a string\"]"))
 	})
 
+	It("marshals strings which should be escaped", func() {
+		result := write(writer, "\\a string")
+		Expect(result).To(Equal("[\"~#'\",\"\\\\a string\"]"))
+	})
```
Comment (thumbnail) [inline writer_test.go:45]: Actually fails when the fix isn't applied, bravo!

**Review by thumbnail — APPROVED** [review]
image

**Pull request merged**